src/main/java/com/zaxxer/hikari/pool/HikariPool.java

```diff
@@ -246,6 +249,7 @@ public synchronized void shutdown() throws InterruptedException
          logPoolState("After shutdown ");
          handleMBeans(this, false);
          metricsTracker.close();
+         hikariPoolLock.unlock();
```

Comment:
Shouldn't this be in a finally block?

Reply:
Yes, you are right. If one of the statements would throw an exception, the lock won't be released.